internal/bsr/bsr.go

```diff
@@ -237,6 +238,173 @@ func OpenSession(ctx context.Context, sessionRecordingId string, f storage.FS, k
+		Name:         sessionRecordingId,
+		Type:         SessionContainer,
+		SubContainer: []Validation{},
+	}
+
+	validationSummary := ValidationSummary{
+		SessionRecordingId: sessionRecordingId,
+		Valid:              true,
+	}
+
+	switch {
+	case sessionRecordingId == "":
+		validationSummary.Valid = false
+		return validation, validationSummary, fmt.Errorf("%s: missing session recording id: %w", op, ErrInvalidParameter)
+	case f == nil:
+		validationSummary.Valid = false
+		return validation, validationSummary, fmt.Errorf("%s: missing storage: %w", op, ErrInvalidParameter)
+	case keyUnwrapFn == nil:
+		validationSummary.Valid = false
+		return validation, validationSummary, fmt.Errorf("%s: missing key unwrap function: %w", op, ErrInvalidParameter)
+	}
+
+	session, err := OpenSession(ctx, sessionRecordingId, f, keyUnwrapFn)
+	if err != nil {
+		validationSummary.Valid = false
+		validationError := fmt.Errorf("%s: failed to retrieve session for %s: %w", op, sessionRecordingId, err)
+		validation.Error = errors.Join(validation.Error, validationError)
+		return validation, validationSummary, validationError
+	}
+
+	// Validate session
+	sessionChecksumValidation, err := session.container.ValidateChecksums(ctx)
+	if err != nil {
+		validationSummary.Valid = false
+		validation.Error = errors.Join(validation.Error, fmt.Errorf("%s: failed to validate session for %s: %w", op, sessionRecordingId, err))
+	}
+
+	validationSummary = updateValidationSummary(ctx, SessionContainer, sessionRecordingId, sessionChecksumValidation, validationSummary)
+
+	validation.ChecksumValidation = sessionChecksumValidation
+
+	conns := session.Meta.connections
+
+	for connId := range conns {
+		// Get connection id
+		connKey := connId
+		lastDotIndex := strings.LastIndex(connId, ".")
+		if lastDotIndex != -1 {
+			connKey = connId[:lastDotIndex]
+		}
+
+		connectionValidation := Validation{
+			Name:         connId,
+			Type:         ConnectionContainer,
+			SubContainer: []Validation{},
+		}
+
+		conn, err := session.OpenConnection(ctx, connKey)
+		if err != nil {
+			validationSummary.Valid = false
+			connectionValidation.Error = errors.Join(connectionValidation.Error, fmt.Errorf("%s: failed to retrieve connection for %s: %w", op, connId, err))
+			continue
+		}
+
+		// Validate current connection
+		connectionChecksumValidation, err := conn.container.ValidateChecksums(ctx)
+		if err != nil {
+			validationSummary.Valid = false
+			connectionValidation.Error = errors.Join(connectionValidation.Error, fmt.Errorf("%s: failed to validate connection for %s: %w", op, connId, err))
+			continue
+		}
+
+		validationSummary = updateValidationSummary(ctx, ConnectionContainer, connId, connectionChecksumValidation, validationSummary)
+
+		connectionValidation.ChecksumValidation = connectionChecksumValidation
+
+		chs := conn.Meta.channels
+
+		for chId := range chs {
+			// Get channel id
+			chsKey := chId
+			lastDotIndex := strings.LastIndex(chId, ".")
```

Comment:
```suggestion
			lastDotIndex := strings.LastIndex(chId, ".channel")
```
Might make this clearer to others in the future